File: 959/C_Hungry_Games.py
#include <bits/stdc++.h>
using namespace std;
#define int long long





void solve() {
   int n,x;
   cin>>n>>x;
   int a[n],pre[n+1]={0},dp[n+3]={0};
   for(int i=0;i<n;++i){
    cin>>a[i];
    pre[i+1]=pre[i]+a[i];
   }
   int ans=0;
   for(int i=n-1;i>=0;--i){
    int val=pre[i]+x;
   
    auto index=lower_bound(pre,pre+n+1,val)-pre;
      if(index==n+1)dp[i]+=(n-i);
       else if(val==pre[index])dp[i]+=(index-i)+dp[index+1];
        else dp[i]+=(index-i-1)+dp[index];
   }
   for(int i=0;i<n+3;++i){
    ans+=dp[i];
   }
   cout<<ans<<endl;
  
}

int32_t main() {
    int t = 1;
    cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}

File: 959/C_Hungry_Games.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long

int32_t main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    int t=1;
    cin >> t;
    while(t--){
        int n, k;
        cin >> n >> k;
        int arr[n];
        for (int i = 0; i < n; i++)
        {
            cin >> arr[i];
        }
        
        int dp[n]={0};
        int j=n-1, i=n-1;
        int sum=0;
        int cnt=0;
        while(j>=0){
            sum+=arr[j];
            if(sum<=k){
                cnt++;
                dp[j]=cnt;
                j--;
            }else{
                cnt++;
                while(sum>k){
                    sum-=arr[i];
                    i--;
                    cnt--;
                }
                dp[j]=cnt;
                j--;
            }
        }

        int ans=dp[n-1];
        for(i=n-2; i>=0; i--){
            if(dp[i]!=0 && i+dp[i]+1<n){
                dp[i]+=dp[i+dp[i]+1];
            }
            dp[i]=max(dp[i], dp[i+1]);
            ans+=dp[i];
        }

        // for(int i=0; i<n; i++){
        //     cout << dp[i] << " ";
        // }
        cout << ans << endl;
    }
    return 0;
}